core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -1019,16 +1019,31 @@ public Builder setBranchSnapshot(long snapshotId, String branch) {
       return this;
     }
 
-    public Builder removeBranch(String branch) {
```

Comment:
Can you keep this and deprecate it? It is technically part of a public-facing API.

Reply:
Ah yeah true. Added a deprecated annotation and commented.